spec/datadog/integration_spec.rb

```diff
@@ -26,18 +26,25 @@ def wait_for_tracer_sent
 
     context 'for threads' do
       let!(:original_thread_count) { thread_count }
+      let!(:original_inspect_threads) { inspect_threads }
 
       def thread_count
         Thread.list.count
       end
 
+      def inspect_threads
+        Thread.list.map.with_index { |t, idx| "#{idx}=#{t.backtrace}" }.join(';')
+      end
+
       it 'closes tracer threads' do
         start_tracer
         wait_for_tracer_sent
 
         shutdown
 
-        expect(thread_count).to eq(original_thread_count)
+        expect(thread_count).to eq(original_thread_count),
+          "Expected #{original_thread_count} threads (#{original_inspect_threads}), "\
+                  "got #{thread_count} threads (#{inspect_threads})"
```

Comment:
This may be slightly flaky/misleading, since `Thread.list` gets called 4 times in this spec, and there's no guarantee that the results are consistent (which may be hella confusing).

What do you think of introducing `let`s for the `Thread.list` before and after, and then using that to obtain the counts and prints? That would ensure that the counts used in the assertion always matched the information we have for debugging.

Reply:
I didn't think that would be an issue, given the test is not time sensitive but state sensitive.
I'll make the changes.

Reply:
I see! My thinking is that the background thread may be some leftover from something else that may still finish at some point, and thus why I think even if the test itself was not sensitive to time, the thread may be.